doc/hardware/porting/arch.rst

```diff
@@ -47,6 +47,16 @@ some are optional:
 * **Linker scripts and toolchains**: architecture-specific details will most
   likely be needed in the build system and when linking the image (required).
 
+* **Memory Management and Memory Mapping**: for architecture-specific details
+  on supporting memory management and memory mapping.
+
+* **Stack Objects**: for architecture-specific details on memory protection
+  hardware regarding stack objects.
+
+* **User Mode Threads**: for supporting threads in user mode.
```

Comment:
These (particularly #1 and #3) feel like they could be more helpful as to *what* should be done. Arguably, not all the other bullet points are doing the best job at it, but they tend to say a few words about what exactly falls in each particular bucket.

Or if there is really no useful additional info to provide, then e.g. leaving things at "User mode threads." would be better than "User mode threads: ... threads in user mode" IMO